### PR TITLE
Save backup import/export location for future import/exports

### LINKED ISSUE #6039: Store backup location
### Checklist

- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### Describe the feature you want

NewPipe should store the backup location in a setting like all other apps do (and just like Newpipe does for downloads).

#### Is your feature request related to a problem? Please describe it

Currently, you have to navigate to the folder where you store your Newpipe backups each time you take a backup or restore one.

#### How will you/everyone benefit from this feature?

Saved time and effort for everyone.

### PR DESCRIPTION
**Pull request by ATofighi**
#### What is it?
- [ ] Bugfix (user facing)
- [x] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

To save location of import/export, I add a new preference key for `import_export_path` and save this preference on result of file chooser activity and send it on intent of file chooser on later tries.

So as a user, If a export data on folder `foo`, when I want to import this data, file chooser show `foo` as start path.

#### Fixes the following issue(s)

- Fixes #6039 

#### APK testing 

On the website the APK can be found by going to the "Checks" tab below the title and then on "artifacts" on the right.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java [hunk 1]
```diff
@@ -239,4 +253,27 @@ private void importDatabase(final String filePath) {
...
         }
     }
+
+    private boolean isValidPath(final String path) {
+        if (path == null || path.isEmpty()) {
+            return false;
+        }
+        final File file = new File(path);
+        return file.exists() && file.isDirectory();
+    }
```
Comment (XiangRongLin) [inline app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java:263]: Can this be moved out of the fragment into a Helper class and unit tests added for it.

File: app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java [hunk 2]
```diff
@@ -41,6 +41,8 @@
 
     private ContentSettingsManager manager;
 
+    private String importExportDataPathKey;
```
Comment (XiangRongLin) [inline app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java:44]: Any reason for it to be an member variable instead of just `getString(R.string.import_export_data_path)` putting whereever its used. I don't see it as a good reason, since it can be easily fetched from that

File: app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java [hunk 3]
```diff
@@ -239,4 +253,27 @@ private void importDatabase(final String filePath) {
...
+        if (!file.isDirectory()) {
+            final File parentFile = file.getParentFile();
+            if (parentFile != null) {
+                directoryPath = parentFile.getAbsolutePath();
+            } else {
+                directoryPath = "";
+            }
+        } else {
+            directoryPath = file.getAbsolutePath();
+        }
```
Comment (XiangRongLin) [inline app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java:276]: I feel like this would be easier to read, if the `!file.isDirectory()` is flipped to `file.isDirectory()` and the bodies are switched accordingly

**Review by XiangRongLin — CHANGES_REQUESTED** [review]
Did you cover the edge case that i mentioned in the issue?

**Comment by ATofighi** [comment]
> Did you cover the edge case that i mentioned in the issue?

Yes, I have tried to cover it in `isValidPath` function. Is there any other edge cases that I must to cover?

File: app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java (hunk 2 above)
Comment (ATofighi) [reply app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java:44]: I have followed older codes that puts `R.string.download_thumbnail_key` and `getString(R.string.youtube_restricted_mode_enabled)` in member variables and I think the code is cleaner with don't repeating `getString(R.string.import_export_data_path)`.

If you disagree with me, I'll fix it when I find free time.

File: app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java (hunk 1 above)
Comment (ATofighi) [reply app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java:263]: Good idea, I'll fix it when I find free time. thanks.

File: app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java (hunk 3 above)
Comment (ATofighi) [reply app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java]: fixed

File: app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java (hunk 1 above)
Comment (ATofighi) [reply app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java]: fixed

File: app/src/test/java/org/schabi/newpipe/util/FilePathHelperTest.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class FilePathHelperTest {
+    @Test
+    public void testIsValidDirectoryPath() throws IOException {
```
Comment (XiangRongLin) [inline app/src/test/java/org/schabi/newpipe/util/FilePathHelperTest.java:14]: Please split up the tests into seperate functions. Basically what is expressed with the comments, can be done with the method name. Which then also gets reflected in the test failure description.

**Review by XiangRongLin — CHANGES_REQUESTED** [review]

**Comment by XiangRongLin** [comment]
>Is there any other edge cases that I must to cover?

Not that i can think of

File: app/src/test/java/org/schabi/newpipe/util/FilePathHelperTest.java (hunk 4 above)
Comment (ATofighi) [reply app/src/test/java/org/schabi/newpipe/util/FilePathHelperTest.java]: fixed

**Comment by XiangRongLin** [comment]
Can you also remove the conversions between file and path in your tests. Like this https://github.com/XiangRongLin/NewPipe/commit/376e5c1546641a10c654475d303b8d298461430a

**Comment by ATofighi** [comment]
> Can you also remove the conversions between file and path in your tests. Like this [XiangRongLin@376e5c1](https://github.com/XiangRongLin/NewPipe/commit/376e5c1546641a10c654475d303b8d298461430a)

Yes, I merged your branch with mine.

**Review by XiangRongLin — APPROVED** [review]
Codewise looks fine to me now.

**Comment by Stypox** [comment]
I don't know how to rebase #5415 on top of this... I think that storing the last used path would require some more work in `saf`.
